app/models/company.rb

```diff
@@ -58,6 +58,8 @@ def last_char_star?
   end
 
   def refresh_search_suggestions
+    # It feels weird that you're calling a class method directly on a concern.
+    # Could this be a service object?
```

Comment:
@NerdDiffer A service object is a on object (quite often a PORO) with the soul objective of mapping a domain _idea_ (ex: MakePayment in an ecommerce app) with domain objects (ex: Payment, AccountBalance, Receipt). A service object removes coupling from domain objects where it does not belong.

Personally, I like a stateless service object. I like the idea that a service object is as near a pure functional interface as possible.

Here's some reading:
https://blog.engineyard.com/2014/keeping-your-rails-controllers-dry-with-services
http://blog.codeclimate.com/blog/2012/10/17/7-ways-to-decompose-fat-activerecord-models/

There are gems that can assist with implementing this pattern. I like to just roll my own service objects. The scaffolding is very minimal and I don't feel the need for a dependency to enforce that pattern